lib/client.js

```diff
@@ -123,6 +123,33 @@ S3Client.prototype.storeFile = function (options, callback) {
 };
 
 /**
+ * Drops file from the underlaying system
+ * @param {Object}   options:
+ *   * filename  {String}  save with this filename
+ *   * headers   {Object}  optional headers to pass
+ * @param {Function} callback <err>
+ */
+S3Client.prototype.dropFile = function (options, callback) {
+    var filename = options.filename,
+        headers  = options.headers || {},
+        s3 = this.s3;
+
+    return new Promise(function dropFilePromise(resolve, reject) {
+        s3.del(filename, headers)
+            .on('response', function dropFileResponse(response) {
+                if (response.statusCode !== 200) {
+                    return reject(new Error('Failed to delete'));
+                }
+                return;
```

Comment:
must call `response.resume()` or `response.pipe` or `response.on('data')` to comsume body and not stall socket open